features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/ProtectedApp.jsx

```diff
@@ -63,8 +67,13 @@ export default class Protected extends Component {
         super(props);
         this.state = {
             settings: null,
+            clientId: Cookies.get(User.CONST.PUBLISHER_CLIENT_ID),
```

Comment:
I think these should be accessed via the User interface

Reply:
for example `User.getClientID` otherwise managing, especially clearing these cookie stores would be hard,

Reply:
These cookies are created in the login_callback.jag and stored indefinitely in the browser, CLIENT_ID cookie value doesn't change often but session_state cookie will be changed after every logout/login. So these cookies are managed by the login and logout process. I've now changed the retrieval of cookies as follows,

` 
clientId: Utils.getCookieWithoutEnvironment(User.CONST.PUBLISHER_CLIENT_ID),
sessionStateCookie:Utils.getCookieWithoutEnvironment(User.CONST.PUBLISHER_SESSION_STATE),`